lib/iris/coords.py

```diff
@@ -1930,28 +1931,37 @@ def _new_points_requirements(self, points):
         Confirm that a new set of coord points adheres to the requirements for
         :class:`~iris.coords.DimCoord` points, being:
             * points are scalar or 1D,
-            * points are numeric, and
+            * points are numeric,
+            * points are not masked, and
             * points are monotonic.
 
         """
         if points.ndim not in (0, 1):
-            raise ValueError(
-                'The points array must be scalar or 1-dimensional.')
+            emsg = 'The {!r} {} points array must be scalar or 1-dimensional.'
+            raise ValueError(emsg.format(self.name(), self.__class__.__name__))
         if not np.issubdtype(points.dtype, np.number):
-            raise ValueError('The points array must be numeric.')
+            emsg = 'The {!r} {} points array must be numeric.'
+            raise ValueError(emsg.format(self.name(), self.__class__.__name__))
+        if ma.isMaskedArray(points) and ma.count_masked(points):
+            emsg = 'A {!r} {} points array must not be masked.'
+            raise TypeError(emsg.format(self.name(), self.__class__.__name__))
```

Comment:
Raise a `TypeError` exceptions here to explicitly catch those masked points. Hence the need for the change [here](https://github.com/SciTools/iris/pull/3427/files#diff-07b1c9cb6b22e14b7f314da7af652aafR394)